packages/starship/build.sh

```diff
@@ -9,11 +9,38 @@ TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_DEPENDS="zlib, openssl"
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--no-default-features --features http"
-TERMUX_PKG_BLACKLISTED_ARCHES="x86_64"
 
 termux_step_pre_configure() {
 	CFLAGS+=" $CPPFLAGS"
 	if [ $TERMUX_ARCH = arm ]; then
 		CFLAGS+=" -fno-integrated-as"
 	fi
+
+	# rust checks libs in PREFIX/lib. It then can't find libc.so and libdl.so because rust program doesn't
+	# know where those are. Putting them temporarly in $PREFIX/lib prevents that failure
+	mv $TERMUX_PREFIX/lib/libz.so.1 $TERMUX_PREFIX/lib/libz.so.1.tmp
+	mv $TERMUX_PREFIX/lib/libz.so $TERMUX_PREFIX/lib/libz.so.tmp
+}
+
+termux_step_post_make_install() {
+	mv $TERMUX_PREFIX/lib/libz.so.1.tmp $TERMUX_PREFIX/lib/libz.so.1
+	mv $TERMUX_PREFIX/lib/libz.so.tmp $TERMUX_PREFIX/lib/libz.so
+}
```

Comment:
Need something better than this. Currently this approach has issue resulting that libz files appear in package because modification timestamp is changing.
```
drwxr-xr-x builder/builder   0 2021-12-17 13:14 ./
drwxr-xr-x builder/builder   0 2021-12-17 13:14 ./data/
drwxr-xr-x builder/builder   0 2021-12-17 13:14 ./data/data/
drwxr-xr-x builder/builder   0 2021-12-17 13:14 ./data/data/com.termux/
drwxr-xr-x builder/builder   0 2021-12-17 13:14 ./data/data/com.termux/files/
drwx------ builder/builder   0 2021-12-17 13:14 ./data/data/com.termux/files/usr/
drwx------ builder/builder   0 2021-12-17 13:14 ./data/data/com.termux/files/usr/bin/
-rwx------ builder/builder 4804272 2021-12-17 13:14 ./data/data/com.termux/files/usr/bin/starship
drwx------ builder/builder       0 2021-12-17 13:14 ./data/data/com.termux/files/usr/share/
drwx------ builder/builder       0 2021-12-17 13:14 ./data/data/com.termux/files/usr/share/doc/
drwx------ builder/builder       0 2021-12-17 13:14 ./data/data/com.termux/files/usr/share/doc/starship/
-rw------- builder/builder     758 2021-12-17 13:14 ./data/data/com.termux/files/usr/share/doc/starship/LICENSE
drwx------ builder/builder       0 2021-12-17 13:14 ./data/data/com.termux/files/usr/lib/
lrwxrwxrwx builder/builder       0 2020-07-04 22:53 ./data/data/com.termux/files/usr/lib/libz.so.1 -> libz.so.1.2.11
lrwxrwxrwx builder/builder       0 2020-07-04 22:53 ./data/data/com.termux/files/usr/lib/libz.so -> libz.so.1.2.11
```

Reply:
This is a function I have in my bashrc file. I used to have a one that backed up timestamps with `stat` and restored them with `touch` (allowing multiple files to be processed), but can't find it. Maybe this will help?

```
function prename(){
  if
    cp --preserve=all "$1" "$2"
  then
    rm -f "$1"
  fi # don't remove files unless copy completed (full disk is bad)
}
```

Reply:
Seems easier to just move them back as done here and then remove them from the package in a massage step, as rust does: https://github.com/termux/termux-packages/blob/master/packages/rust/build.sh#L77, https://github.com/termux/termux-packages/blob/master/packages/rust/build.sh#L95